packages/flutter/lib/src/rendering/sliver.dart

```diff
@@ -1311,6 +1313,26 @@ abstract class RenderSliver extends RenderObject {
   @override
   void handleEvent(PointerEvent event, SliverHitTestEntry entry) { }
 
+  /// Whether the [SemanticsNode]s associated with this [RenderSliver] should
+  /// be excluded from the semantic scrolling area.
+  ///
+  /// [RenderSliver]s that stay on the screen even though the user has scrolled
+  /// past them (e.g. a pinned app bar) should set this to `true`.
+  set excludeFromSemanticsScrolling(bool value) {
+    if (_excludeFromSemanticsScrolling == value)
+      return;
+    _excludeFromSemanticsScrolling = value;
+    markNeedsSemanticsUpdate();
```

Comment:
I think you can pass both onlyChanges:true and noGeometry:true here, if instead of addTag you had setTag and it took a boolean. Then your semantics annotator could always be set (wouldn't have to ever be null), and you could just set or not set the tag according to the state of this value.